src/main/java/com/github/wojtechm/zadania_rekrutacyjne/string/ParenthesesCombinations.java

```diff
@@ -18,7 +20,58 @@
 @Difficulty(Level.HARD)
 class ParenthesesCombinations {
 
+    private ParenthesesCombinations(){}
+
     static List<String> parenthesesCombinations(int pairs) {
-        return null;
+        if (pairs <= 0){
+            return Collections.emptyList();
+        }
+
+        ArrayList<String> out = new ArrayList<>();
+        int number = 0;
+        for (int i = pairs; i > 0; i--) {
+            number += 1 << i * 2 - 1;
+        }
+
+        int maxNumber = (int) Math.pow(2, pairs * 2.0) - 1;
+        int mask = 0;
+        for (int i = 0; i < pairs; i++) {
+            mask += (int) Math.pow(2, i);
+        }
+        maxNumber &= mask << pairs;
+        int maxBeforeNumber = (int) (maxNumber - Math.pow(2, pairs - 1));
+
+        makeOut(pairs, out, number, maxBeforeNumber);
```

Comment:
One problem here
Google phrase "to make out" and check the definition. Then change the method name to something more appropriate ;)